File: .github/taskIdbyowner.py
import os
import json

codeowners_file_path = '/workspaces/azure-pipelines-tasks/.github/CODEOWNERS'
team_name = '@microsoft/azure-artifacts-packages'

def extract_tasks_owned_by_team(file_path, team):
    tasks = []
    with open(file_path, 'r') as file:
        for line in file:
            if team in line:
                task_path = line.split()[0]
                tasks.append(task_path)
    return tasks

def extract_task_id(task_path):
    task_json_path = os.path.join(task_path, 'task.json')
    if os.path.exists(task_json_path):
        with open(task_json_path, 'r') as file:
            task_data = json.load(file)
            return task_data.get('id')
    return None

# Specify the path to the CODEOWNERS file and the team name


# Extract tasks owned by the specified team
tasks_owned_by_team = extract_tasks_owned_by_team(codeowners_file_path, team_name)

# Extract and print the task IDs for each task
for task in tasks_owned_by_team:
    task_id = extract_task_id(task)
    if task_id:
        print(f'Task Path: {task}, Task ID: {task_id}')
    else:
        print(f'Task Path: {task}, Task ID: Not Found')    

File: .github/taskIdbyowner.js
// /workspaces/azure-pipelines-tasks $ node ./.github/taskIdbyowner.js

const fs = require('fs');
const path = require('path');

const codeownersFilePath = '/workspaces/azure-pipelines-tasks/.github/CODEOWNERS';
const teamName = '@microsoft/azure-artifacts-packages';

function extractTasksOwnedByTeam(filePath, team) {
    const tasks = [];
    const lines = fs.readFileSync(filePath, 'utf-8').split('\n');
    lines.forEach(line => {
        if (line.includes(team)) {
            const taskPath = line.split(/\s+/)[0];
            tasks.push(taskPath);
        }
    });
    return tasks;
}

function extractTaskId(taskPath) {
    const taskJsonPath = path.join(taskPath, 'task.json');
    if (fs.existsSync(taskJsonPath)) {
        const taskData = JSON.parse(fs.readFileSync(taskJsonPath, 'utf-8'));
        return taskData.id || null;
    }
    return null;
}

// Extract tasks owned by the specified team
const tasksOwnedByTeam = extractTasksOwnedByTeam(codeownersFilePath, teamName);

// Extract and print the task IDs for each task
tasksOwnedByTeam.forEach(task => {
    const taskId = extractTaskId(task);
    if (taskId) {
        console.log(`Task Path: ${task}, Task ID: ${taskId}`);
    } else {
        console.log(`Task Path: ${task}, Task ID: Not Found`);
    }
});